config/env_sample.hjson

```diff
@@ -252,7 +253,34 @@
                 and membership.id is not null 
                 and REGEXP_EXTRACT(`group`.id, r'.*:(.*)') IN UNNEST(@course_ids)
                 '''
-        }
+        },
+        "canvas_pages": {
+            "query":
+                '''
+                SELECT 'canvas_pages' AS resource_type,
+                REPLACE(JSON_EXTRACT_SCALAR(extensions_json, '$[\'com.instructure.canvas\'][request_url]'), 'https://example.instructure.com/courses', '') as resource_id,
```

Comment:
I am not sure why the query chose an URL as the resource id. The id contains a long page name, which can be not palatable for eyes and when doing the join this could be problematic. example: `/1233/pages/podcast-nice-white-parents-episode-2-i-still-believe-in-it?module_item_id=12333`

Each page is associated with a page id and we should use that instead. For example

```
"object":{
      "id":"urn:instructure:canvas:wikiPage:17700000000111111",
      "type":"Page",
      "extensions":{
         "com.instructure.canvas":{
            "asset_name":"Podcast: Nice White Parents, Episode 2 \u201cI Still Believe in It\u201d ",
            "asset_type":"wiki_page",
            "entity_id":"17700000000111111",
            "context_account_id":"17700000000000111",
            "http_method":"GET"
         }
```

Is there any reason for having the resource id is page URL instead of page id?

Reply:
I tested the approach of having a page_id instead of page_URL as the resource_id it will fit the cron design. But One exception, when sending the url in the response need to construct the URL with the course id, I believe that might require a change to cron.py 

`"resource_name":"https:\/\/\/example.instructure.com\/courses\/1111111\/pages\/1770000000022222|xyg page name|fas fa-file-code fa-lg"`

So based on the resource the URL formatting might change. There might be better way of fixing this, but this is what I can think of

Reply:
I like the idea of use page id, but I think it is hard to implement alongside with other resource types. See my comments below.

Reply:
@pushyamig please check the latest query format. I have updated it based on your suggestion.